packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1128,10 +1128,12 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
       onChunk: loadingBuilder == null ? null : _handleImageChunk,
       onError: widget.errorBuilder != null
         ? (dynamic error, StackTrace stackTrace) {
-            setState(() {
-              _lastException = error;
-              _lastStack = stackTrace;
-            });
+            if (mounted) {
```

Comment:
Something is weird here, when the image widget disposed, it should remove the listener from the image stream. This should not have been called

Reply:
Ok so the issue is here https://github.com/flutter/flutter/blob/32547dcc7ee36a9d387dbf2d111b21a12b8192be/packages/flutter/lib/src/widgets/image.dart#L1130

The _getListener is used for removing the listener on dispose
https://github.com/flutter/flutter/blob/32547dcc7ee36a9d387dbf2d111b21a12b8192be/packages/flutter/lib/src/widgets/image.dart#L1190

This is buggy because the remove listener uses `==` to figure out which listener to remove. If you look at the equal override in image stream
https://github.com/flutter/flutter/blob/32547dcc7ee36a9d387dbf2d111b21a12b8192be/packages/flutter/lib/src/painting/image_stream.dart#L125
However, if you have a errorBuilder, it creates an anonymous function every time getListener is called so it fail the equal check. The correct fix should be makes getListener return the ImageStreamListener that passes the equal check